lusSTR/marker.py

```diff
@@ -99,13 +103,18 @@ def flankseq_5p(self):
         if self.uas:
             return None
         front, back = self._uas_bases_to_trim()
+        if front == 0:
+            return ''
         return self.sequence[:front]
 
     @property
     def flank_5p(self):
-        if self.uas:
+        if self.uas or self.flankseq_5p == '':
             return None
-        elif self.kit == 'powerseq' and self.data['Power_5'] > self.data['Foren_5']:
+        elif (
+            self.kit == 'powerseq' and self.data['Power_5'] > self.data['Foren_5']
+            and self.data['Foren_5'] > 0
+        ):
```

Comment:
It's hard to look at this and nod my head and say "Ok, yeah, I understand what's going on here." Like, why is it *this particular set of conditions* that makes collapsing flanks in this way suitable for these loci?

The code does little to illuminate the reader's understanding of why particular loci are handled a certain way. To be clear, I don't think this is a reflection the coder. I understand this whole business of locus-specific conventions is very complex and fraught with substantial baggage and difficult to design for. Sometimes it seems like there are more exceptions than rules! If the code is providing the correct annotations, that may be the best we can expect at the moment.